WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldEditorView.swift

```diff
@@ -53,54 +32,64 @@ struct CustomFieldEditorView: View {
+                        .subheadlineStyle()
                         .padding(insets: Layout.inputInsets)
                         .background(Color(.listForeground(modal: false)))
                         .overlay(
                             RoundedRectangle(cornerRadius: Layout.cornerRadius).stroke(Color(.separator))
                         )
-                        .onChange(of: key) { _ in
-                            checkForModifications()
-                        }
                 }
 
                 // Value Input
                 VStack(alignment: .leading, spacing: Layout.subSectionsSpacing) {
                     HStack {
-                        Text("Value") // todo-13493: set String to be translatable
+                        Text(Localization.valueLabel)
                             .foregroundColor(Color(.text))
                             .subheadlineStyle()
 
                         Spacer()
 
                         if !isReadOnlyValue {
-                            Button(action: {
-                                showRichTextEditor = true
-                            }) {
-                                Text("Edit in Rich Text Editor") // todo-13493: set String to be translatable
-                                    .font(.footnote)
+                            Picker(Localization.editorPickerLabel, selection: $showRichTextEditor) {
+                                Text(Localization.editorPickerText).tag(false)
+                                Text(Localization.editorPickerHTML).tag(true)
                             }
-                            .buttonStyle(.plain)
-                            .foregroundColor(Color(uiColor: .accent))
+                            .pickerStyle(.segmented)
+                            .fixedSize()
                         }
                     }
 
-                    TextEditor(text: isReadOnlyValue ? .constant(value) : $value)
-                        .bodyStyle()
+                    if showRichTextEditor {
+                        AztecEditorView(initialValue: value,
+                                        onContentChanged: { content in
+                            DispatchQueue.main.async {
+                                value = content
+                            }
+                        })
                         .frame(minHeight: Layout.minimumEditorSize)
+                        .clipped()
                         .padding(insets: Layout.inputInsets)
                         .background(Color(.listForeground(modal: false)))
                         .overlay(
                             RoundedRectangle(cornerRadius: Layout.cornerRadius).stroke(Color(.separator))
                         )
-                        .onChange(of: value) { _ in
-                            checkForModifications()
-                        }
+                    } else {
+                        TextEditor(text: isReadOnlyValue ? .constant(value) : $value)
```

Comment:
Since there's a placeholder text on both the key field and the HTML mode, should we add placeholder text for the text mode as well?

Reply:
So unlike in Key field where having a value is required (although the checking logic is not yet added here), having empty content in Value field is a supported feature of Custom Fields. In this case, we want the empty content to be displayed as-is, instead of displaying a placeholder message.

This is also the same as in Core:

<img width="654" alt="Screenshot 2024-09-24 at 19 49 36" src="https://github.com/user-attachments/assets/67bc5d8f-837c-4da5-9b66-46c9818cea94">